Test in-cluster console for e2e tests

spadgett opened the pull request:
Previously we were running the bridge binary locally. Instead test the in-cluster console, which tests the built console image.

/cc @jcaianirh @alecmerdler @benjaminapetersen 

https://jira.coreos.com/browse/CONSOLE-1267
https://jira.coreos.com/browse/CONSOLE-1266

benjaminapetersen commented:
```
level=info msg="Waiting up to 30m0s for the cluster to initialize..."
level=fatal msg="failed to initialize the cluster: Cluster operator network has not yet reported success"
```

benjaminapetersen commented:
/retest

spadgett commented:
/retest

```
level=info msg="Creating cluster..."
level=error
level=error msg="Error: Error applying plan:"
level=error
level=error msg="1 error occurred:"
level=error msg="\t* module.vpc.aws_route_table_association.route_net[3]: 1 error occurred:"
level=error msg="\t* aws_route_table_association.route_net.3: timeout while waiting for state to become 'success' (timeout: 5m0s)"
```

spadgett commented:
```
+ ./test-gui.sh e2e
+ copyArtifacts
++ pwd
+ echo 'Copying artifacts from /go/src/github.com/openshift/console...'
Copying artifacts from /go/src/github.com/openshift/console...
+ cp -rv ./frontend/gui_test_screenshots /tmp/artifacts/gui_test_screenshots
cp: cannot stat './frontend/gui_test_screenshots': No such file or directory
```

Not sure what failed in this run. It looks like the e2e tests didn't start.

I added a guard in `copyArtifacts` to avoid that error and added `set -x` to `test-gui.sh` for better logging.

spadgett commented:
> level=fatal msg="failed to initialize the cluster: Cluster operator network has not yet reported success"

/retest

spadgett commented:
/retest

spadgett commented:
Looks like the call to `yarn run webdriver-update` is failing. We send output to `/dev/null` so it's not clear why.

spadgett commented:
Ah, I forgot to add `yarn install` since we removed `build-frontend.sh`

alecmerdler commented:
/test console-e2e

spadgett commented:
/retest

spadgett commented:
/retest

spadgett commented:
/refresh

spadgett commented:
/retest

benjaminapetersen commented on frontend/integration-tests/protractor.conf.ts:
```diff
@@ -67,6 +67,9 @@ export const config: Config = {
     });
 
+    const url = await browser.getCurrentUrl();
+    console.log('Last browser URL', url);
```
+1

spadgett commented:
/hold

spadgett commented:
/retest

spadgett commented:
/refresh

spadgett commented:
/test console-e2e

spadgett commented:
Closing and reopening to kick the tests

spadgett commented:
/retest

spadgett commented:
/retest
/hold cancel

spadgett commented:
/retest

spadgett commented:
/retest

spadgett commented:
/retest

spadgett commented:
/retest

spadgett commented:
/refresh

spadgett commented:
/refresh

spadgett commented:
/refresh

spadgett commented:
/retest

spadgett commented:
/retest

spadgett commented:
/retest

spadgett commented:
OK we had one login succeed with a different e2e flake and the last failure got to the login redirect.

> Last browser URL https://console-openshift-console.apps.ci-op-psn4k3pg-514ba.origin-ci-int-aws.dev.rhcloud.com/auth/login

/retest

spadgett commented:
Login failed due to bug 1679272. We might need to fold #1206 and this into one PR.

> Last browser URL https://console-openshift-console.apps.ci-op-psn4k3pg-514ba.origin-ci-int-aws.dev.rhcloud.com/error?error=invalid_code&error_type=auth

/retest

spadgett commented:
/retest

```
Failed Specs:

1) Performance test : downloads new bundle for operator-management
   Error: Failed expectation
       at Object.<anonymous> /go/src/github.com/openshift/console/frontend/integration-tests/tests/performance.scenario.ts:96:30
       at step /go/src/github.com/openshift/console/frontend/integration-tests/tests/performance.scenario.ts:32:23
       at Object.next /go/src/github.com/openshift/console/frontend/integration-tests/tests/performance.scenario.ts:13:53
       at fulfilled /go/src/github.com/openshift/console/frontend/integration-tests/tests/performance.scenario.ts:4:58
       at process._tickCallback internal/process/next_tick.js:68:7
   TypeError: Cannot read property 'decodedBodySize' of null
       at Object.<anonymous> /go/src/github.com/openshift/console/frontend/integration-tests/tests/performance.scenario.ts:101:36
       at step /go/src/github.com/openshift/console/frontend/integration-tests/tests/performance.scenario.ts:32:23
       at Object.next /go/src/github.com/openshift/console/frontend/integration-tests/tests/performance.scenario.ts:13:53
       at fulfilled /go/src/github.com/openshift/console/frontend/integration-tests/tests/performance.scenario.ts:4:58
       at process._tickCallback internal/process/next_tick.js:68:7
   From: Task: Run it"downloads new bundle for operator-management" in control flow
       at UserContext.<anonymous> /go/src/github.com/openshift/console/frontend/node_modules/jasminewd2/index.js:94:19
   From asynchronous test: 
   Error
       at /go/src/github.com/openshift/console/frontend/integration-tests/tests/performance.scenario.ts:86:5
       at /go/src/github.com/openshift/console/frontend/node_modules/immutable/dist/immutable.js:2699:43
       at List.__iterate /go/src/github.com/openshift/console/frontend/node_modules/immutable/dist/immutable.js:2206:13
       at OrderedMap.__iterate /go/src/github.com/openshift/console/frontend/node_modules/immutable/dist/immutable.js:2698:25
       at OrderedMap.forEach /go/src/github.com/openshift/console/frontend/node_modules/immutable/dist/immutable.js:4381:19
       at Suite.<anonymous> /go/src/github.com/openshift/console/frontend/integration-tests/tests/performance.scenario.ts:78:17
       at Env.<anonymous> /go/src/github.com/openshift/console/frontend/node_modules/jasmine-fail-fast/dist/jasmine-fail-fast.js:57:26
```

spadgett commented:
> level=fatal msg="failed to initialize the cluster: Cluster operator network has not yet reported success"

/retest

spadgett commented:
Failed because of bug 1679272 (needs #1206)

/retest

spadgett commented:
Failed again because of 1679272

/retest

spadgett commented:
I opened #1207 which combines the two fixes to unblock the queue

spadgett commented:
/retest

enj commented on frontend/integration-tests/protractor.conf.ts:
```diff
@@ -31,6 +31,7 @@ export const config: Config = {
   plugins: [failFast.init()],
   capabilities: {
     browserName: 'chrome',
+    acceptInsecureCerts : true,
```
What certs does this use?  Could it be having the same SA crt changes under it issues?

spadgett replied:
This is to bypass the browser warning for self-signed certificates.

enj commented on test-prow-e2e.sh:
```diff
@@ -2,23 +2,27 @@
 
-./build.sh
+export BRIDGE_AUTH_USERNAME=kubeadmin
+# don't log kubeadmin-password
```
:smile:

enj commented on test-prow-e2e.sh:
```diff
@@ -2,23 +2,27 @@
+set +x
+export BRIDGE_AUTH_PASSWORD="$(cat "${INSTALLER_DIR}/auth/kubeadmin-password")"
+set -x
+export BRIDGE_BASE_ADDRESS="https://$(oc get route console -n openshift-console -o jsonpath='{.spec.host}')"
```
At some point you should be able to get this from your top level config

spadgett commented:
Same performance test failure. Something is different when running in cluster:

```
 Failed Specs:

1) Performance test : downloads new bundle for operator-management
   Error: Failed expectation
       at Object.<anonymous> /go/src/github.com/openshift/console/frontend/integration-tests/tests/performance.scenario.ts:96:30
       at step /go/src/github.com/openshift/console/frontend/integration-tests/tests/performance.scenario.ts:32:23
       at Object.next /go/src/github.com/openshift/console/frontend/integration-tests/tests/performance.scenario.ts:13:53
       at fulfilled /go/src/github.com/openshift/console/frontend/integration-tests/tests/performance.scenario.ts:4:58
       at process._tickCallback internal/process/next_tick.js:68:7
   TypeError: Cannot read property 'decodedBodySize' of null
       at Object.<anonymous> /go/src/github.com/openshift/console/frontend/integration-tests/tests/performance.scenario.ts:101:36
       at step /go/src/github.com/openshift/console/frontend/integration-tests/tests/performance.scenario.ts:32:23
       at Object.next /go/src/github.com/openshift/console/frontend/integration-tests/tests/performance.scenario.ts:13:53
       at fulfilled /go/src/github.com/openshift/console/frontend/integration-tests/tests/performance.scenario.ts:4:58
       at process._tickCallback internal/process/next_tick.js:68:7
   From: Task: Run it"downloads new bundle for operator-management" in control flow
       at UserContext.<anonymous> /go/src/github.com/openshift/console/frontend/node_modules/jasminewd2/index.js:94:19
   From asynchronous test: 
   Error
       at /go/src/github.com/openshift/console/frontend/integration-tests/tests/performance.scenario.ts:86:5
       at /go/src/github.com/openshift/console/frontend/node_modules/immutable/dist/immutable.js:2699:43
       at List.__iterate /go/src/github.com/openshift/console/frontend/node_modules/immutable/dist/immutable.js:2206:13
       at OrderedMap.__iterate /go/src/github.com/openshift/console/frontend/node_modules/immutable/dist/immutable.js:2698:25
       at OrderedMap.forEach /go/src/github.com/openshift/console/frontend/node_modules/immutable/dist/immutable.js:4381:19
       at Suite.<anonymous> /go/src/github.com/openshift/console/frontend/integration-tests/tests/performance.scenario.ts:78:17
       at Env.<anonymous> /go/src/github.com/openshift/console/frontend/node_modules/jasmine-fail-fast/dist/jasmine-fail-fast.js:57:26
```

spadgett commented:
/retest

kyoto commented:
/retest

kyoto commented:
/retest

kyoto commented:
/retest

spadgett commented:
/retest

spadgett commented:
/retest

spadgett commented:
/retest

spadgett commented:
/retest

spadgett commented:
/retest

spadgett commented:
/retest

spadgett commented:
/retest

spadgett commented:
/retest

spadgett commented:
Got all the way to the etcd operator tests

```
Interacting with the etcd OCS creates etcd Operator `Deployment` 1m41s
Error: Timeout - Async callback was not invoked within timeout specified by jasmine.DEFAULT_TIMEOUT_INTERVAL.
```

/retest

spadgett commented:
/retest

spadgett commented:
/retest

kyoto commented:
/retest

kyoto commented:
/retest

kyoto commented:
/retest

kyoto commented:
/retest

kyoto commented:
/retest

kyoto commented:
/retest

kyoto commented:
/retest

spadgett commented:
@kyoto Thanks for helping me retest this 👍

spadgett commented:
@alecmerdler OLM etcd scenario has failed 3 times. Can you help? Wondering if there's a real bug here:

https://openshift-gce-devel.appspot.com/build/origin-ci-test/pr-logs/pull/openshift_console/1197/pull-ci-openshift-console-master-console-e2e/673/

https://openshift-gce-devel.appspot.com/build/origin-ci-test/pr-logs/pull/openshift_console/1197/pull-ci-openshift-console-master-console-e2e/675/

https://openshift-gce-devel.appspot.com/build/origin-ci-test/pr-logs/pull/openshift_console/1197/pull-ci-openshift-console-master-console-e2e/678/

I don't see the etcd deployment in the screenshots:

https://storage.googleapis.com/origin-ci-test/pr-logs/pull/openshift_console/1197/pull-ci-openshift-console-master-console-e2e/673/artifacts/console-e2e/gui_test_screenshots/463125ea270e98ddf6d549a19bd5c35f.png

Pod logs should be in the artifacts if we need them. I might try to increase the timeout in case we're just not waiting long enough.

spadgett commented:
/retest

spadgett commented:
Rate limited

/retest

```
level=fatal msg="failed to fetch Terraform Variables: failed to fetch dependency of \"Terraform Variables\": failed to fetch dependency of \"Bootstrap Ignition Config\": failed to fetch dependency of \"Common Manifests\": failed to generate asset \"DNS Config\": getting public zone for \"origin-ci-int-aws.dev.rhcloud.com\": listing hosted zones: Throttling: Rate exceeded\n\tstatus code: 400, request id: 329198a7-36af-11e9-8cbf-abfcb2fdb5d0"
```

spadgett commented:
/retest

spadgett commented:
/refresh

spadgett commented:
backend/frontend failures were infrastructure flakes (test-bin pod was evicted)

spadgett commented:
/retest

spadgett commented:
/test backend
/test frontend

spadgett commented:
/refresh

spadgett commented:
/test backend
/test frontend

spadgett commented:
/test backend

spadgett commented:
/refresh
